pages/api/discover/popular.js

```diff
@@ -16,25 +19,29 @@ export default async function handler(req, res) {
 
   const directoryPath = path.join(process.cwd(), "data");
 
-  const fullPopularProfiles = popularProfiles.flatMap((profile) => {
-    const filePath = path.join(directoryPath, `${profile.username}.json`);
-    try {
-      const user = JSON.parse(fs.readFileSync(filePath, "utf8"));
-
-      if (user.displayStatsPublic) {
-        return {
-          ...user,
-          ...profile._doc,
-        };
+  const fullPopularProfiles = await Promise.all(
+    popularProfiles.flatMap(async (profile) => {
+      const filePath = path.join(directoryPath, `${profile.username}.json`);
+      try {
+        const user = await memoizedReadAndParseJsonFile(filePath);
+
+        if (user.displayStatsPublic) {
+          return {
+            ...user,
+            ...profile._doc,
+          };
+        }
+
+        return {};
```

Comment:
These are filtered out by the `flatMap`

Reply:
For some reason the [] array is not being filtered out by the flatMap! maybe I should just use map

Reply:
You are returning `{}` so it won't get filtered out, if you ever the code back to `[]` it will filter them out and then no need for the filtering out later on

Reply:
I changed it to [] in my dev env and it wasnt being filtered out